gnovm/pkg/doc/json_doc.go

```diff
@@ -113,32 +143,126 @@ func (d *Documentable) WriteJSONDocumentation() (*JSONDocumentation, error) {
 	for _, fun := range pkg.Funcs {
 		jsonDoc.Funcs = append(jsonDoc.Funcs, &JSONFunc{
 			Name:      fun.Name,
-			Signature: mustFormatNode(d.pkgData.fset, fun.Decl),
+			Crossing:  isCrossing(fun),
+			Signature: mustFormatNode(d.pkgData.fset, fun.Decl, opt.Source, file),
 			Doc:       string(pkg.Markdown(fun.Doc)),
 			Params:    d.extractJSONFields(fun.Decl.Type.Params),
 			Results:   d.extractJSONFields(fun.Decl.Type.Results),
 		})
 	}
 
 	for _, typ := range pkg.Types {
+		// Set typeSpec to the ast.TypeSpec within the declaration that defines the symbol.
+		var typeSpec *ast.TypeSpec
+		for _, spec := range typ.Decl.Specs {
+			tSpec := spec.(*ast.TypeSpec) // Must succeed
+			if typ.Name == tSpec.Name.Name {
+				typeSpec = tSpec
+				break
+			}
+		}
+		if typeSpec == nil || typeSpec.Type == nil {
+			// We don't expect this
+			continue
+		}
+		typeExpr := typeSpec.Type
+
+		// De-parenthesize
+		for {
+			if t, ok := typeExpr.(*ast.ParenExpr); ok {
+				typeExpr = t.X
+			} else {
+				break
+			}
+		}
```

Comment:
Helper?

Reply:
Done in https://github.com/gnolang/gno/pull/3931/commits/ab603802f5553d834a44f4b67b8ff208f87d14ca . This only works for one element inside the parentheses. Is it possible for multiple elements in the parentheses like `(x, y)` ?

Reply:
Won't worry about `(x, y)` for now.